### PR TITLE
Allowing SGE schedulers to use the plugin

### LINKED ISSUE #40: Serial is always assumed in `start_voro`
When running `start_voro` the serial keyword is not explicitly passed via `get_inputs_voronoi`, this in turn results in the `get_inputs_common` setting an option resource dictionary that looks like this
```
        # overwrite settings for serial run
        options['withmpi'] = False
        options['resources'] = {"num_machines": 1}
```

This will fail if one is using a SGE kind of scheduler for the machine where voronoi is compiled.

the simplest solution would be to set

`serial = not self.ctx.withmpi`

```
builder = get_inputs_voronoi(
                    voronoicode, structure, options, label, description, params=params, serial=serial)
```

Unless there is a specific reason for this @PhilippRue , @broeder-j ?

I recognize that in principle one can have voronoi and the host code in two different machines, with two different kinds of schedulers and setups. So perhaps this is a too restrictive approach.

### PR DESCRIPTION
**Pull request by JPchico**
The plugin could not work as the `get_inputs_common` added submission options which are incompatible for the SGE scheduler. Now it checks which scheduler is provided and sets options which correspond to a serial simulation. Fixes #40

### CODE REVIEW

File: aiida_kkr/tools/common_workfunctions.py [hunk 1]
```diff
@@ -331,7 +337,9 @@ def get_inputs_common(calculation, code, remote, structure, options, label, desc
 
     if code:
         inputs.code = code
-
+        _sched = load_computer(code.computer.label).scheduler_type
```
Comment (PhilippRue) [inline aiida_kkr/tools/common_workfunctions.py:340]: Why is `load_computer` needed? Does `code.computer` not give you the computer already?

File: examples/kkr_submit_scf.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        }
+    )
+
+    # Setting parameters for the submission of the job
+    options = {
+        'resources': {
+            # Total number of mpi processes
+            'tot_num_mpiprocs': 16,
+            # Name of the parallel environment
+            'parallel_env': 'mpi',
```
Comment (PhilippRue) [inline examples/kkr_submit_scf.py:79]: Maybe you can put a comment here to tell that this is for a SGE scheduler?

**Review by PhilippRue — CHANGES_REQUESTED** [review]
Thanks @JPchico. I left two minor comments. The rest looks fine to me.

File: aiida_kkr/tools/common_workfunctions.py (hunk 1 above)
Comment (JPchico) [reply aiida_kkr/tools/common_workfunctions.py:340]: Yeah, you are right, I copied this from some other function where I wanted to load the computer for some tests. I have changed it as you recommed.

File: examples/kkr_submit_scf.py (hunk 2 above)
Comment (JPchico) [reply examples/kkr_submit_scf.py:79]: Sure thing, added now a check for whether the code is in an SGE or SLURM/PBS machine. And added some reasonable options

**Comment by JPchico** [comment]
I have added the changes that you requested @PhilippRue , plus some smaller formatting issues.

**Review by PhilippRue — APPROVED** [review]
Thanks @JPchico. Everything looks good for me

**Comment by PhilippRue** [comment]
@JPchico I will merge this in tomorrow if there is nothing to add

**Comment by JPchico** [comment]
@PhilippRue I think this is fine, I have not seen any other issue cropping up in my tests.

**Pull request merged**